admin/admin.go

```diff
@@ -19,15 +21,38 @@ type AdminServer struct {
 	cancel context.CancelFunc
 	wg     sync.WaitGroup
 
+	rpcServer     *rpc.Server // New field for the RPC server
+	networkConfig *config.NetworkConfig
+
 	port uint64
 }
 
-func NewAdminServer(log log.Logger, port uint64) *AdminServer {
-	return &AdminServer{log: log, port: port}
+type RPCMethods struct {
+	Log           log.Logger
+	NetworkConfig *config.NetworkConfig
+}
+
+func NewAdminServer(log log.Logger, port uint64, networkConfig *config.NetworkConfig) *AdminServer {
+	return &AdminServer{log: log, port: port, networkConfig: networkConfig}
 }
 
 func (s *AdminServer) Start(ctx context.Context) error {
 	router := setupRouter()
+
+	// Set up RPC server
+	rpcServer := rpc.NewServer()
+	rpcMethods := &RPCMethods{
+		Log:           s.log,
+		NetworkConfig: s.networkConfig, // Ensure this is correctly set
+	}
+
+	if err := rpcServer.RegisterName("admin", rpcMethods); err != nil {
+		return fmt.Errorf("failed to register RPC methods: %w", err)
+	}
+
+	// Mount the RPC server on the root path `/` to handle all RPC requests
+	router.Any("/", gin.WrapH(rpcServer))
```

Comment:
we should stick to just POST requests here

`router.POST`

Reply:
done✅